deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c

```diff
@@ -83,16 +153,38 @@ static inline bool _find_row_index_in_Ah
 	return false ;
 }
 
+// Create a new iterator
+GrB_Info GxB_MatrixTupleIter_new
+(
+	GxB_MatrixTupleIter **iter,     // iterator to create
+	const GrB_Matrix A              // matrix to iterate over
+) {
+	GB_WHERE(A, "GxB_MatrixTupleIter_new (iter, A)") ;
+	GB_RETURN_IF_NULL_OR_FAULTY(A) ;
+	int sparsity_type;
+	GxB_Matrix_Option_get(A, GxB_SPARSITY_CONTROL, &sparsity_type) ;
+	if(sparsity_type != GxB_SPARSE && sparsity_type != GxB_HYPERSPARSE)
+		GB_ERROR (GrB_INVALID_VALUE, "Invalid sparsity type: %d", sparsity_type) ;
+
+	*iter = GB_MALLOC(1, GxB_MatrixTupleIter) ;
```

Comment:
would be a bit nicer if
`GxB_MatrixTupleIter *it = GB_MALLOC(1, GxB_MatrixTupleIter) ;`
and at the end of this function we set input argument:
`*iter = it;`